### PR TITLE
ddl, br: improve compatibility for PiTR and ingest ddl jobs

### PR DESCRIPTION
**Pull request by joccau**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tidb/issues/38045

Problem Summary:
The log-backup task(PITR) has compatibility problem with the DDL add index with the ingest method

### What is changed and how it works?
- If the DDL `add index` with ingest method is running, creating log-backup task will fail.
- If the log-backup task exists, the DDL `add index` will run by normal(old and slow) method.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

### Test report
[Test about compatibility for PITR and DDL-add index with ingest method](https://pingcap.feishu.cn/docx/doxcn9d0JzmlNVzsnf7SQW0Dpkg)

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: br/pkg/task/stream.go [hunk 1]
```diff
@@ -496,6 +497,9 @@ func RunStreamStart(
 		return errors.New("Unable to create task about log-backup. " +
 			"please set TiKV config `log-backup.enable` to true and restart TiKVs.")
 	}
+	if !ddl.IngestJobsClear(se.GetSessionCtx()) {
+		return errors.Annotate(berrors.ErrUnknown, "Unable to create log backup task. please clear DDL about injest job.")
```
Comment (tangenta) [inline br/pkg/task/stream.go:501]:
```suggestion
		return errors.Annotate(berrors.ErrUnknown, "Unable to create log backup task. Please wait until the DDL tasks are finished.")
```

File: ddl/index.go [hunk 2]
```diff
@@ -718,7 +719,31 @@ func canUseIngest(w *worker) bool {
 		failpoint.Return(true)
 	})
 	// Ingest way is not compatible with PiTR.
-	return !utils.CheckLogBackupEnabled(ctx)
+	return !utils.IsLogBackupUsage(ctx)
+}
+
+func IngestJobsClear(ctx sessionctx.Context) bool {
+	sess := session{ctx}
+	template := "select job_meta from mysql.tidb_ddl_job where reorg and (type = %d or type = %d);"
```
Comment (tangenta) [inline ddl/index.go:727]:
```suggestion
	template := "select job_meta from mysql.tidb_ddl_job where reorg and (type = %d or type = %d) and processing;"
```

**Review by tangenta — APPROVED** [review]

File: ddl/index.go [hunk 3]
```diff
@@ -718,7 +719,31 @@ func canUseIngest(w *worker) bool {
...
 	// Ingest way is not compatible with PiTR.
-	return !utils.CheckLogBackupEnabled(ctx)
+	return !utils.IsLogBackupUsage(ctx)
+}
+
+func IngestJobsClear(ctx sessionctx.Context) bool {
+	sess := session{ctx}
+	template := "select job_meta from mysql.tidb_ddl_job where reorg and (type = %d or type = %d) and processing;"
+	sql := fmt.Sprintf(template, model.ActionAddIndex, model.ActionAddPrimaryKey)
+	rows, err := sess.execute(context.Background(), sql, "")
```
Comment (tangenta) [inline ddl/index.go:729]:
```suggestion
	rows, err := sess.execute(context.Background(), sql, "check-pitr")
```

File: br/pkg/utils/db.go [hunk 4]
```diff
@@ -112,3 +112,7 @@ func LogBackupTaskCountDec() {
 func CheckLogBackupTaskExist() bool {
 	return logBackupTaskCount > 0
 }
+
+func IsLogBackupUsage(ctx sessionctx.Context) bool {
```
Comment (3pointer) [inline br/pkg/utils/db.go:116]: maybe IsLogBackupInUse?

Comment (joccau) [reply br/pkg/utils/db.go]: Done

File: ddl/index.go [hunk 5]
```diff
@@ -718,7 +719,32 @@ func canUseIngest(w *worker) bool {
...
+	}
+	for _, row := range rows {
+		jobBinary := row.GetBytes(0)
+		runJob := model.Job{}
+		err := runJob.Decode(jobBinary)
+		if err != nil {
+			logutil.BgLogger().Warn("cannot check ingest job", zap.Error(err))
+			return false
+		}
+		if runJob.ReorgMeta.ReorgTp == model.ReorgTypeLitMerge {
```
Comment (3pointer) [inline ddl/index.go:743]: please add some comments here

**Review by 3pointer — COMMENTED** [review]
rest LGTM

File: ddl/index.go (hunk 5 above)
Comment (tangenta) [inline ddl/index.go:743]:
```suggestion
		// Check whether this add index job is using lighting to do the backfill work.
		if runJob.ReorgMeta.ReorgTp == model.ReorgTypeLitMerge {
```

**Review by 3pointer — APPROVED** [review]

**Pull request merged**